"make all" doesn't run black if lint fails

As of ctsm5.1.dev163, `make lint` errors:
```
...
Your code has been rated at 8.80/10 (previous run: 8.80/10, +0.00)

make: *** [Makefile:50: lint] Error 30
```

This means that, when calling `make all`, the `black` test isn't reached.

Doing either of the following in `python/Makefile` fixes the issue:
1. Add `--fail-under=0` to `PYLINT_ARGS`
2. Change the order of the tests in `all` from `test lint black` to `test black lint`.

--- a/python/Makefile
+++ b/python/Makefile
@@ -27,7 +27,7 @@ PYLINT_SRC = \
 #       ../cime_config/buildlib \
 #       ../cime_config/buildnml
 
-all: test lint black
+all: test black lint
 	@echo
 	@echo
 	@echo "Successfully ran all standard tests"

--- a/python/Makefile
+++ b/python/Makefile
@@ -19,7 +19,7 @@ ifneq ($(verbose), not-set)
 endif
 
 PYLINT=pylint
-PYLINT_ARGS=-j 4 --rcfile=ctsm/.pylintrc
+PYLINT_ARGS=-j 4 --rcfile=ctsm/.pylintrc --fail-under=0
 PYLINT_SRC = \
 	ctsm
 #  NOTE: These don't pass pylint checking and should be added when we put into effort to get them to pass

